### PR TITLE
fix: integrated PBJ error propagation

### PR DESCRIPTION
**Pull request by mattp-swirldslabs**
**Description**:
* Added integration with PBJ fix #304 

**Related issue(s)**:

Fixes PBJ ticket #304 

**Notes for reviewer**:

**Checklist**

- [X] Tested (unit, integration, etc.)

### CODE REVIEW

File: server/src/main/java/com/hedera/block/server/consumer/ConsumerStreamResponseObserver.java [hunk 1]
```diff
@@ -118,7 +118,17 @@ public void onEvent(
...
+                try {
+                    responseSender.send(subscribeStreamResponse);
+                } catch (IllegalArgumentException e) {
+                    throw e;
+                } catch (RuntimeException e) {
+                    isResponsePermitted.set(false);
+                    subscriptionHandler.unsubscribe(this);
+                    LOGGER.log(
+                            DEBUG,
+                            "RuntimeException caught from Pipeline instance. Unsubscribed ConsumerBlockItemObserver instance");
```
Comment (AlfredoG87) [inline server/src/main/java/com/hedera/block/server/consumer/ConsumerStreamResponseObserver.java:130]: should this be DEBUG? I get is not a actual `error` but maybe a warning?

**Review by AlfredoG87 — APPROVED** [review]
LGTM, just one small question.

**Review by ata-nas — DISMISSED** [review]
Looks good, one nit and one question.

File: server/src/main/java/com/hedera/block/server/consumer/ConsumerStreamResponseObserver.java (hunk 1 above)
Comment (mattp-swirldslabs) [reply server/src/main/java/com/hedera/block/server/consumer/ConsumerStreamResponseObserver.java:130]: The exceptions we're catching here will be a downstream client hitting control-c 99% of the time

File: server/src/main/java/com/hedera/block/server/consumer/ConsumerStreamResponseObserver.java [hunk 2]
```diff
@@ -118,7 +118,17 @@ public void onEvent(
...
                 final SubscribeStreamResponseUnparsed subscribeStreamResponse = event.get();
                 final ResponseSender responseSender = getResponseSender(subscribeStreamResponse);
-                responseSender.send(subscribeStreamResponse);
+                try {
+                    responseSender.send(subscribeStreamResponse);
+                } catch (IllegalArgumentException e) {
+                    throw e;
+                } catch (RuntimeException e) {
+                    isResponsePermitted.set(false);
+                    subscriptionHandler.unsubscribe(this);
```
[the remark this answers is not in the document]
Comment (mattp-swirldslabs) [reply server/src/main/java/com/hedera/block/server/consumer/ConsumerStreamResponseObserver.java]: I consolidated the logic and added some comments.

File: server/src/main/java/com/hedera/block/server/consumer/ConsumerStreamResponseObserver.java (hunk 1 above)
Comment (mattp-swirldslabs) [reply server/src/main/java/com/hedera/block/server/consumer/ConsumerStreamResponseObserver.java]: I added more fine-grained exception handling. Now the UncheckedIOExceptions will get logged to DEBUG and any other RuntimeException should show up as an ERROR.

**Review by AlfredoG87 — APPROVED** [review]
LGTM!

**Review by ata-nas — APPROVED** [review]
Looks good @mattp-swirldslabs! Resolved from my side.

**Pull request merged**